be/src/runtime/runtime_state.cpp

```diff
@@ -147,7 +147,10 @@ Status RuntimeState::init(const TUniqueId& fragment_instance_id, const TQueryOpt
         _query_options.max_errors = 100;
     }
 
-    if (_query_options.batch_size <= 0) {
+    // if BE was in grayscale upgrade. old BE chunk size was 4094.
+    // if FE set a zero batch_size, batch_size will be set to DEFAULT_CHUNK_SIZE
+    // (DEFAULT_CHUNK_SIZE was 2048 before version 2.0/2.1.0). which will cause overflow
+    if (_query_options.batch_size <= DEFAULT_CHUNK_SIZE) {
```

Comment:
If this batch_size is greater than 4096, new version be will use the greater chunk size, and the older BE will use 4096. IMO it will also cause the overflow problem. Am I right?

Reply:
yes.

Reply:
if FE set batch_size greater than DEFAULT_CHUNK_SIZE or old BE set vector_chunk_size, it will also cause a overflow. so I suggest add this to RELEASE NOTES.